typeset/converter.js

```diff
@@ -239,7 +244,43 @@ async function injectCoverageCollection (page) {
   }
 }
 
+const highlightCodeElements = (log, inputPath) => {
+  const highlightPath = inputPath.replace('baked', 'highlight')
+
+  if (!fileExists.sync(inputPath)) {
+    log.error(`Input XHTML file not found: "${inputPath}"`)
+    return STATUS_CODE.ERROR
+  }
+
+  const data = fs.readFileSync(inputPath, 'utf8')
+  const inputFile = data.toString()
+
+  const dom = new JSDOM(inputFile, {
+    contentType: 'application/xhtml+xml'
+  })
+
+  const preTagElements = [...dom.window.document.querySelectorAll('pre')]
```

Comment:
Rather than selecting all `pre` tags this could just select the ones with `data-lang` since the other ones would not be converted anyway.